docker/tensorflow-application/video-serving/streaming.py

```diff
@@ -0,0 +1,61 @@
+from flask import Flask, Response, request
+from flask import render_template, send_file
+import threading
+import argparse
+import datetime
+import imutils
+import time
+import cv2
+import requests
+import numpy as np
+from PIL import Image
+from flask import jsonify
+import time
+
+
+outputFrame = None 
+lock = threading.Lock()
+
+app = Flask(__name__, template_folder='templates')
+
+@app.route("/")
+def index():
+	return render_template('index.html')
+
+def stream_to_memory():
+	global outputFrame, lock	
+	list1 = request.json['instances']
+	for l in list1:
+		img=np.asarray(l, dtype=np.uint32)
+		outputFrame = Image.fromarray(img.astype('uint8'))
+
+
+@app.route("/video")
+def video():
+	def produce_frame(): 
+		global outputFrame, lock 
+		
+		while True: 
+			if outputFrame is None: 
+				continue
+			
+			image1 = np.asarray(outputFrame)
+			(flag, image) = cv2.imencode(".JPEG", image1)
+
+			if not flag:
+				continue
+			
+			yield(b'--frame\r\n' b'Content-Type: image/jpeg\r\n\r\n' + 
+				bytearray(image) + b'\r\n')
+			time.sleep(1.0)
```

Comment:
It should probably be less than a sec. I assume the tensor flow application is producing more than 1 image a second, correct? Do you know it's rate? I'd probably do something like .2 seconds or similar.

Reply:
I added this delay since the TensorFlow inference takes a lot of time on my machine. We'll have to tweak this number so by the time the image batches received are displayed on the browser, the next batch is ready to be sent to the video server. This is just an attempt to mask the delay between the batches unless we use something like a cache.

Reply:
Worse case scenario if your streamer is faster than your analysis is that the same image would be sent multiple times, right? Which will look exactly the same to the end user as just sending images slower. So using a smaller delay will always work better because if the analysis is faster the stream will be too. If it's not the stream will look the same, so no downside.